www/scripts/login/login.js

```diff
@@ -0,0 +1,219 @@
+// -------------------------------------------------------------------------
+//                     The CodeChecker Infrastructure
+//   This file is distributed under the University of Illinois Open Source
+//   License. See LICENSE.TXT for details.
+// -------------------------------------------------------------------------
+
+define([
+  'dojo/_base/declare',
+  'dojo/cookie',
+  'dojo/dom',
+  'dojo/dom-construct',
+  'dojo/dom-class',
+  'dojo/keys',
+  'dojo/on',
+  'dijit/form/Button',
+  'dijit/form/TextBox',
+  'dijit/layout/BorderContainer',
+  'dijit/layout/ContentPane',
+  'codechecker/MessagePane',
+  'codechecker/hashHelper'],
+function (declare, cookie, dom, domConstruct, domClass, keys, on, Button,
+  TextBox, BorderContainer, ContentPane, MessagePane, hash) {
+
+  // A stripped-down version of the "normal" CodeChecker GUI header, tailored
+  // for a lightweight login window.
+  var HeaderPane = declare(ContentPane, {
+    postCreate : function () {
+      this.inherited(arguments);
+
+      //--- Logo ---//
+
+      var logoContainer = domConstruct.create('div', {
+        id : 'logo-container'
+      }, this.domNode);
+
+      var logo = domConstruct.create('span', { id : 'logo' }, logoContainer);
+
+      var logoText = domConstruct.create('div', {
+        id : 'logo-text',
+        innerHTML : 'CodeChecker'
+      }, logoContainer);
+    }
+  });
+
+  var LoginPane = declare(ContentPane, {
+    _doLogin : function() {
+      try {
+        var sessionToken = CC_AUTH_SERVICE.performLogin(
+          'Username:Password', this.txtUser.value + ':' + this.txtPass.value);
+
+        domClass.add(this._mbox.domNode, 'mbox-success');
+        this._mbox.show("Successfully logged in!", '');
+
+        // Set the cookie in the browser.
+        cookie(CC_AUTH_COOKIE_NAME, sessionToken, { path : '/' });
+
+        var returnTo = hash.getState('returnto') || '';
+        window.location = window.location.origin + '/' + returnTo;
+      } catch (exc) {
+        domClass.add(this._mbox.domNode, 'mbox-error');
+        this._mbox.show("Failed to log in!", exc.message);
+
+        this.txtUser.set('value', '');
+        this.txtPass.set('value', '');
+        this.txtUser.focus();
+      }
+    },
+
+    constructor : function() {
+      this._mbox = new MessagePane({
+        class   : 'mbox'
+      });
+
+      this.txtUser = new TextBox({
+        class : 'form-input',
+        name  : 'username'
+      });
+
+      this.txtPass = new TextBox({
+        class : 'form-input',
+        name  : 'password',
+        type  : 'password'
+      });
+
+      var that = this;
+      this.btnSubmit = new Button({
+        label   : "Login",
+        onClick : function () {
+          that._doLogin();
```

Comment:
If I click on the login button the authentication takes too much time. It would be great to use a dojo stand by (https://dojotoolkit.org/reference-guide/1.10/dojox/widget/Standby.html) to show that the operation is in process.

Reply:
First comment fixed.

Second I will look into later, no time to do right now. Also, should the authentication really be that slow?